eventlet/websocket.py

```diff
@@ -427,7 +426,7 @@ def _parse_messages(self):
                 buf = buf[end_idx + 1:]
             elif frame_type == 255:
                 # Closing handshake.
-                assert six.indexbytes(buf, 1) == 0, "Unexpected closing handshake: %r" % buf
+                assert buf[1] == 0, "Unexpected closing handshake: %r" % buf
```

Comment:
I would suggest taking advantage of the situation to replace this “assertion” with a condition. This is a module loaded at runtime, so if optimization is used this code will be removed.

Else it should be rework with a follow up patch.

Reply:
From a hasty look at the context, what you're saying makes sense.

Asserts are for cases where the matched condition indicates your own code is incorrectly written. They "cannot happen" because as everyone knows, we ourselves write bug-free code, so why check it.

This code  *appears* to be parsing external inputs, which means other people's code is buggy and we have to check that :D or well, innumerable things including fuzzer attacks.

I can fix it with a follow-up patch.

Reply:
thanks